packages/swayze/lib/src/widgets/table_body/selections/fill_selections/fill_selection.dart

```diff
@@ -0,0 +1,270 @@
+  final SelectionBorder border;
+  final Size size;
+  final Offset offset;
+
+  const _AnimatedFillSelection({
+    Key? key,
+    required this.border,
+    required this.size,
+    required this.offset,
+    required Duration duration,
+  }) : super(key: key, duration: duration);
+
+  @override
+  _AnimatedSelectionState createState() => _AnimatedSelectionState();
+}
+
+class _AnimatedSelectionState
+    extends AnimatedWidgetBaseState<_AnimatedFillSelection> {
+  late final viewportContext = ViewportContextProvider.of(context);
+
+  Tween<double>? _left;
+  Tween<double>? _top;
+  SizeTween? _size;
+
+  @override
+  void forEachTween(TweenVisitor<dynamic> visitor) {
+    _left = visitor(
+      _left,
+      widget.offset.dx,
+      (dynamic value) => Tween<double>(begin: value as double),
+    ) as Tween<double>?;
+
+    _top = visitor(
+      _top,
+      widget.offset.dy,
+      (dynamic value) => Tween<double>(begin: value as double),
+    ) as Tween<double>?;
+
+    _size = visitor(
+      _size,
+      widget.size,
+      (dynamic value) => SizeTween(begin: value as Size),
+    ) as SizeTween?;
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final animation = this.animation;
+    final left = _left?.evaluate(animation) ?? 0;
+    final top = _top?.evaluate(animation) ?? 0;
+    final size = _size?.evaluate(animation) ?? Size.zero;
+
+    return FillSelectionPainter(
+      border: widget.border,
+      offset: Offset(left, top),
+      size: size,
+    );
+  }
+}
+
+/// A [LeafRenderObjectWidget] that render a fill selection.
+@visibleForTesting
+class FillSelectionPainter extends LeafRenderObjectWidget {
+  final SelectionBorder border;
+  final Offset offset;
+  final Size size;
+
+  const FillSelectionPainter({
+    required this.border,
+    required this.offset,
+    required this.size,
+  });
+
+  @override
+  _RenderFillSelectionPainter createRenderObject(BuildContext context) {
+    return _RenderFillSelectionPainter(
+      border,
+      offset,
+      size,
+    );
+  }
+
+  @override
+  void updateRenderObject(
+    BuildContext context,
+    _RenderFillSelectionPainter renderObject,
+  ) {
+    renderObject
+      ..border = border
+      ..offset = offset
+      ..definedSize = size;
+  }
+}
+
+class _RenderFillSelectionPainter extends RenderBox {
+  _RenderFillSelectionPainter(
+    this._border,
+    this._offset,
+    this._definedSize,
+  );
+
+  Offset _offset;
+  Offset get offset => _offset;
+  set offset(Offset value) {
+    if (_offset != value) {
+      _offset = value;
+      markNeedsPaint();
+    }
+  }
+
+  Size _definedSize;
+  Size get definedSize => _definedSize;
+  set definedSize(Size value) {
+    if (_definedSize != value) {
+      _definedSize = value;
+      markNeedsLayout();
+    }
+  }
+
+  SelectionBorder _border;
+  SelectionBorder get border => _border;
+  set border(SelectionBorder value) {
+    if (_border != value) {
+      _border = value;
+      markNeedsPaint();
```

Comment:
we're only setting paint because we assume the width doesnt change?  Or is it ok to only paint even if the width of the border changes?

Reply:
The size is independent from the border size and offset, that's why we just paint a new one. If the size changes, we have to re-layout, if not, we can just repaint.